webui/src/lib/api/index.js

```diff
@@ -55,6 +55,22 @@ export const defaultAPIHeaders = {
     "X-Lakefs-Client": "lakefs-webui/__buildVersion",
 };
 
+export const parseRawHeaders = (rawHeaders) => {
+    const headersString = typeof rawHeaders === 'string' ? rawHeaders : rawHeaders.toString();
+    const cleanedHeadersString = headersString.trim();
+    const headerLines = cleanedHeadersString.split('\n');
+    const parsedHeaders = headerLines.reduce((acc, line) => {
+        let [key, ...value] = line.split(':'); // split into key and the rest of the value
+        key = key.trim(); 
+        value = value.join(':').trim(); 
+        if (key && value) {
+            acc[key.toLowerCase()] = value;
+        }
+        return acc;
+    }, {});
+    return parsedHeaders;
+};
+
```

Comment:
moved the func to this file since it is being used here, and ive noticed that funcs from here are exported to the other file and not the other way around.